### DISCUSSION TITLE
Can i use quinoa when making a native build?

### DISCUSSION THREAD

**Discussion by keesjetil** [quarkiverse/quarkus-quinoa#740] [General]
Good day fellow developers,

I trying to work with quinoa and angular 17 in quarkus, and so far in dev mode and prod mode its working like a charm.

However when trying to write a native application using this tutorial: https://quarkus.io/guides/writing-native-applications-tips i cant seem to get the frontend working, always seeing a screen with resources not found. setting the quarkus.native.resources.includes to the build dir or to the generated target dir where the contents of the build dir are moved to did not work. I still see resource not found.

After much experimentation with maven, i found a workaround by using the frontend maven plugin to build my app and the maven ant plugin to move the resources to the META-INF dir in the pre-package phase. This is a very inefficient workaround since at the package phase the app is being build again.

long story short, is there an efficient way to configure quinoa such that native builds serve the build resources in a correct manor?

Tips and tricks are much appreciated.

With kind regards,

Kees Til

**Comment by melloware** [comment]
Quinoa works with Native build out of the box do you have a small Sample Quarkus reproducer you can show that its not working?  I have 3 different apps all building Native Images with Quinoa just fine.

**Reply by keesjetil** [reply]
not at the moment, how did you fix your application.properties or pom.xml to use the build-dir or target dir where quinoa resources are preent https://quarkus.io/guides/writing-native-applications-tips in your application.properties?

```
Quarkus automatically includes the resources present in META-INF/resources (the web resources) but, outside this directory, you are on your own.
```

maybe i have conflicting properties in application.properties?

**Reply by melloware** [reply]
you can study my full working Quinoa demo which uses real database, rest services etc to serve a Quinoa app based on React.  https://github.com/melloware/quarkus-primereact

**Reply by melloware** [reply]
Why are you resources outside of `META-INF/` ???

**Reply by keesjetil** [reply]
Because quinoa moves them see ==> https://docs.quarkiverse.io/quarkus-quinoa/dev/config-reference.html

the [quarkus.quinoa.build-dir] and [quarkus.quinoa.just-build] move the files to the 'target/quinoa/build' dir. on top of that if you see the warning on the getting started page ==> https://docs.quarkiverse.io/quarkus-quinoa/dev/index.html the resources are not meant to go into the META-INF folder,at least that is what im seeing if i build my app.

i am looking at your project right now and just looking at it (not building yet), i do not see a reason why your generated resources would be in a META-INF folder. I will inspect your project later on today thanks for the inspiration, it might help me understanding what is going wrong.

**Reply by melloware** [reply]
Ok let me know. It should all just work.  What language and build tool are you using?  Vite? React? angular?

**Reply by melloware** [reply]
Unzip this: 
[quinoa-quickstart.zip](https://github.com/user-attachments/files/16737593/quinoa-quickstart.zip)

Run `mvn quarkus:dev` and go to http://localhost:8080

Its as bare bones a Quinoa app as you will find.  Then build it for native.

**Reply by SteffanVerlaan** [reply]
I'm running into the same issue. I used your bare bones setup with but with an Angular frontend. The resource not found comes up when setting the quarkus.http.root-path property.
I've tried a whole bunch of configurations in Angular as well to try and get this to work, but without success.

**Reply by melloware** [reply]
ahhh it might be Angular specific @SteffanVerlaan did my example with a simple Vite frontend work in Native?

**Reply by SteffanVerlaan** [reply]
just tried it with your vite frontend, that works fine out the box, but the same resource not found when setting quarkus.http.root-path

**Reply by melloware** [reply]
@SteffanVerlaan can you look through these tickets it looks like we have 1 open and 15 closed tickets related to `root-path` can you check the open or closed ones to see if they are your issue?

https://github.com/quarkiverse/quarkus-quinoa/issues?q=is%3Aissue%20state%3Aclosed%20quarkus.http.root-path

**Reply by SteffanVerlaan** [reply]
it looks an awful lot like these issues:
https://github.com/quarkiverse/quarkus-quinoa/issues/867
https://github.com/quarkiverse/quarkus-quinoa/issues/960

I've tried with a bunch of different quarkus and quinoa versions, including 3.25.3 referenced as a fix in issue 867

looks like the html is now served correctly, but the assets aren't resolving to the path, they're still at root level

**Reply by melloware** [reply]
ok maybe create a small Quarkus reproducer  with Angular showing the issue and opening a ticket?  Keep the reproducer as small as possible.
